src/atoms_calculators.jl

```diff
@@ -0,0 +1,106 @@
+#
+# Define AtomsCalculators interface for DFTK.
+#
+# This interface is inspired by the one used in Molly.jl, 
+# see https://github.com/JuliaMolSim/Molly.jl/blob/master/src/types.jl
+#
+# TODO: Decide how Hamiltonian terms should be updated upon change of the positions.
+# These terms are held inside the `basis`. Currently, the whole basis is rebuilt.
+# TODO: Find out if symmetries can be re-enabled (see issue on GH).
+
+struct DFTKState
+    scfres::NamedTuple
+end
+function construct_dummy_state(basis)
+    ρ = guess_density(basis)
+    ψ = nothing # Will get initialized by SCF.
+    return DFTKState((;ρ, ψ, basis))
+end
+
+struct DFTKParameters
+    Ecut::Real
+    kgrid::Union{Nothing,<:AbstractVector{Int}}
+    tol::Real
+    temperature::Real
+end
+
+mutable struct DFTKCalculator <: AbstractCalculator
+    params::DFTKParameters
+    scf_callback
+    state::DFTKState
+end
+
+function prepare_basis(system::AbstractSystem, params::DFTKParameters)
+    model = model_LDA(system; temperature=params.temperature, symmetries=false)
+    basis = PlaneWaveBasis(model; params.Ecut, params.kgrid)
+    return basis
+end
+
+function DFTKCalculator(system; Ecut::Real, kgrid::Union{Nothing,<:AbstractVector{Int}}, tol=1e-6,
+        temperature=zero(Real), verbose_scf=false, state=nothing
+    )
```

Comment:
Here have only a low-level constructor for now taking the dicts (basically can be done using the `Base.@kwdef`). I agree it's not a bad idea to set `callback=identity` by default in the `scf_kwargs`. If you want you can also make a high-level constructor a bit like the one you have here that maps down to the low-level one.